### ISSUE TITLE
`-Werror=maybe-uninitialized` raised in `zig_llvm-ar.cpp`

### ISSUE THREAD

**Issue by nektro** [ziglang/zig#14409]
### Zig Version

562d52e23dce703b0eb45a212522588274afbd38

### Steps to Reproduce and Observed Behavior

```
cmake .. -DZIG_STATIC_ZLIB=ON -DZIG_NO_LIB=ON -DCMAKE_PREFIX_PATH='/home/meghan/src/llvm-15/out' -DZIG_STATIC_LLVM=OFF
make install
```

```
-- Configuring zig version 0.11.0-dev.1404+562d52e23
-- Configuring done
-- Generating done
-- Build files have been written to: /home/meghan/src/zig/build
Consolidate compiler generated dependencies of target zigcpp
[  5%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_llvm-ar.cpp.o
/home/meghan/src/zig/src/zig_llvm-ar.cpp: In function ‘void performWriteOperation(ArchiveOperation, llvm::object::Archive*, std::unique_ptr<llvm::MemoryBuffer>, std::vector<llvm::NewArchiveMember>*)’:
/home/meghan/src/zig/src/zig_llvm-ar.cpp:1084:71: error: ‘Kind’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
 1084 |                    Kind, Deterministic, Thin, std::move(OldArchiveBuf));
      |                                                                       ^
/home/meghan/src/zig/src/zig_llvm-ar.cpp: In function ‘int ar_main(int, char**)’:
/home/meghan/src/zig/src/zig_llvm-ar.cpp:476:5: error: ‘Operation’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
  476 |     if (Operation != Extract)
      |     ^~
/home/meghan/src/zig/src/zig_llvm-ar.cpp:332:20: note: ‘Operation’ was declared here
  332 |   ArchiveOperation Operation;
      |                    ^~~~~~~~~
cc1plus: all warnings being treated as errors
make[2]: *** [CMakeFiles/zigcpp.dir/build.make:90: CMakeFiles/zigcpp.dir/src/zig_llvm-ar.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:91: CMakeFiles/zigcpp.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

```
[nix-shell:~/src/zig/build]$ ll `which $CC`
-r-xr-xr-x 1 root root 7489 Dec 31  1969 /nix/store/lsyzzki1iv9gwk4vdss7i1cjxrnxling-gcc-wrapper-11.3.0/bin/gcc
```

### Expected Behavior

successful build

**Comment by nektro** [comment]
adding this diff leads to the following log

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 662bd09cd..0669aac7f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -707,7 +707,7 @@ if(ZIG_STATIC)
 endif()
 
 add_library(zigcpp STATIC ${ZIG_CPP_SOURCES})
-set_target_properties(zigcpp PROPERTIES COMPILE_FLAGS ${EXE_CXX_FLAGS})
+set_target_properties(zigcpp PROPERTIES COMPILE_FLAGS "${EXE_CXX_FLAGS} -Wno-error=maybe-uninitialized")
 
 target_link_libraries(zigcpp LINK_PUBLIC
     ${CLANG_LIBRARIES}
```

```
[nix-shell:~/src/zig/build]$ make install
-- Configuring zig version 0.11.0-dev.1404+562d52e23
-- Configuring done
-- Generating done
-- Build files have been written to: /home/meghan/src/zig/build
Consolidate compiler generated dependencies of target zigcpp
[  5%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_llvm.cpp.o
[ 10%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_llvm-ar.cpp.o
/home/meghan/src/zig/src/zig_llvm-ar.cpp: In function ‘void performWriteOperation(ArchiveOperation, llvm::object::Archive*, std::unique_ptr<llvm::MemoryBuffer>, std::vector<llvm::NewArchiveMember>*)’:
/home/meghan/src/zig/src/zig_llvm-ar.cpp:1084:71: warning: ‘Kind’ may be used uninitialized in this function [-Wmaybe-uninitialized]
 1084 |                    Kind, Deterministic, Thin, std::move(OldArchiveBuf));
      |                                                                       ^
/home/meghan/src/zig/src/zig_llvm-ar.cpp: In function ‘int ar_main(int, char**)’:
/home/meghan/src/zig/src/zig_llvm-ar.cpp:476:5: warning: ‘Operation’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  476 |     if (Operation != Extract)
      |     ^~
/home/meghan/src/zig/src/zig_llvm-ar.cpp:332:20: note: ‘Operation’ was declared here
  332 |   ArchiveOperation Operation;
      |                    ^~~~~~~~~
[ 15%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_clang.cpp.o
[ 20%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_clang_driver.cpp.o
[ 25%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_clang_cc1_main.cpp.o
[ 30%] Building CXX object CMakeFiles/zigcpp.dir/src/zig_clang_cc1as_main.cpp.o
/home/meghan/src/zig/src/zig_clang_cc1as_main.cpp: In function ‘bool ExecuteAssemblerImpl({anonymous}::AssemblerInvocation&, clang::DiagnosticsEngine&)’:
/home/meghan/src/zig/src/zig_clang_cc1as_main.cpp:548:23: warning: ‘Value’ may be used uninitialized in this function [-Wmaybe-uninitialized]
  548 |     Ctx.setSymbolValue(Parser->getStreamer(), Sym, Value);
      |     ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[ 35%] Building CXX object CMakeFiles/zigcpp.dir/src/windows_sdk.cpp.o
[ 40%] Linking CXX static library zigcpp/libzigcpp.a
[ 40%] Built target zigcpp
[ 45%] Building C object CMakeFiles/zig-wasm2c.dir/stage1/wasm2c.c.o
[ 50%] Linking C executable zig-wasm2c
[ 50%] Built target zig-wasm2c
[ 55%] Converting /home/meghan/src/zig/stage1/zig1.wasm to /home/meghan/src/zig/build/zig1.c
[ 60%] Building C object CMakeFiles/zig1.dir/zig1.c.o
[ 65%] Building C object CMakeFiles/zig1.dir/stage1/wasi.c.o
[ 70%] Linking C executable zig1
[ 70%] Built target zig1
[ 75%] Running zig1.wasm to produce /home/meghan/src/zig/build/zig2.c
[ 80%] Running zig1.wasm to produce /home/meghan/src/zig/build/compiler_rt.c
[ 85%] Building C object CMakeFiles/zig2.dir/zig2.c.o
In file included from /nix/store/ab85b675p235dncb1s6hsb9pcckyxafj-glibc-2.34-210-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/ab85b675p235dncb1s6hsb9pcckyxafj-glibc-2.34-210-dev/include/limits.h:26,
                 from /nix/store/zg9fzn6m0480xqr2mx5ynqkziribycgv-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/limits.h:203,
                 from /nix/store/zg9fzn6m0480xqr2mx5ynqkziribycgv-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/syslimits.h:7,
                 from /nix/store/zg9fzn6m0480xqr2mx5ynqkziribycgv-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/limits.h:34,
                 from /home/meghan/src/zig/lib/zig.h:5,
                 from /home/meghan/src/zig/build/zig2.c:1:
/nix/store/ab85b675p235dncb1s6hsb9pcckyxafj-glibc-2.34-210-dev/include/features.h:412:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
/home/meghan/src/zig/build/zig2.c: In function ‘crypto_blake3_Output_chainingValue’:
/home/meghan/src/zig/build/zig2.c:2572406: warning: ‘crypto_blake3_CompressGeneric_compress’ reading 64 bytes from a region of size 32 [-Wstringop-overread]
2572406 |  t11 = crypto_blake3_CompressGeneric_compress(t2, t4, t6, t8, t10);
        | 
/home/meghan/src/zig/build/zig2.c:2572406: note: referencing argument 2 of type ‘const zig_u32 * const’ {aka ‘const unsigned int * const’}
/home/meghan/src/zig/build/zig2.c:2733384: note: in a call to function ‘crypto_blake3_CompressGeneric_compress’
2733384 | static zig_T_tuple_7b_5b16_5du32_7d_60048 crypto_blake3_CompressGeneric_compress(zig_u32 const a0[zig_const_arr zig_as_u64(8)], zig_u32 const a1[zig_const_arr zig_as_u64(16)], zig_u32 const a2, zig_u64 const a3, zig_u8 const a4) {
        | 
/home/meghan/src/zig/build/zig2.c: In function ‘crypto_blake3_ChunkState_update’:
/home/meghan/src/zig/build/zig2.c:2572584: warning: ‘crypto_blake3_CompressGeneric_compress’ reading 64 bytes from a region of size 32 [-Wstringop-overread]
2572584 |     t27 = crypto_blake3_CompressGeneric_compress(t17, t11, zig_as_u32(64), t4, t26);
        | 
/home/meghan/src/zig/build/zig2.c:2572584: note: referencing argument 2 of type ‘const zig_u32 * const’ {aka ‘const unsigned int * const’}
/home/meghan/src/zig/build/zig2.c:2733384: note: in a call to function ‘crypto_blake3_CompressGeneric_compress’
2733384 | static zig_T_tuple_7b_5b16_5du32_7d_60048 crypto_blake3_CompressGeneric_compress(zig_u32 const a0[zig_const_arr zig_as_u64(8)], zig_u32 const a1[zig_const_arr zig_as_u64(16)], zig_u32 const a2, zig_u64 const a3, zig_u8 const a4) {
        | 
/home/meghan/src/zig/build/zig2.c: In function ‘crypto_blake3_Output_rootOutputBytes’:
/home/meghan/src/zig/build/zig2.c:2572739: warning: ‘crypto_blake3_CompressGeneric_compress’ reading 64 bytes from a region of size 32 [-Wstringop-overread]
2572739 |    t19 = crypto_blake3_CompressGeneric_compress(t9, t11, t13, t18, t17);
        | 
/home/meghan/src/zig/build/zig2.c:2572739: note: referencing argument 2 of type ‘const zig_u32 * const’ {aka ‘const unsigned int * const’}
/home/meghan/src/zig/build/zig2.c:2733384: note: in a call to function ‘crypto_blake3_CompressGeneric_compress’
2733384 | static zig_T_tuple_7b_5b16_5du32_7d_60048 crypto_blake3_CompressGeneric_compress(zig_u32 const a0[zig_const_arr zig_as_u64(8)], zig_u32 const a1[zig_const_arr zig_as_u64(16)], zig_u32 const a2, zig_u64 const a3, zig_u8 const a4) {
        | 
[ 90%] Building C object CMakeFiles/zig2.dir/compiler_rt.c.o
In file included from /nix/store/ab85b675p235dncb1s6hsb9pcckyxafj-glibc-2.34-210-dev/include/bits/libc-header-start.h:33,
                 from /nix/store/ab85b675p235dncb1s6hsb9pcckyxafj-glibc-2.34-210-dev/include/limits.h:26,
                 from /nix/store/zg9fzn6m0480xqr2mx5ynqkziribycgv-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/limits.h:203,
                 from /nix/store/zg9fzn6m0480xqr2mx5ynqkziribycgv-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/syslimits.h:7,
                 from /nix/store/zg9fzn6m0480xqr2mx5ynqkziribycgv-gcc-11.3.0/lib/gcc/x86_64-unknown-linux-gnu/11.3.0/include-fixed/limits.h:34,
                 from /home/meghan/src/zig/lib/zig.h:5,
                 from /home/meghan/src/zig/build/compiler_rt.c:1:
/nix/store/ab85b675p235dncb1s6hsb9pcckyxafj-glibc-2.34-210-dev/include/features.h:412:4: warning: #warning _FORTIFY_SOURCE requires compiling with optimization (-O) [-Wcpp]
  412 | #  warning _FORTIFY_SOURCE requires compiling with optimization (-O)
      |    ^~~~~~~
[ 95%] Linking CXX executable zig2
[ 95%] Built target zig2
[100%] Building stage3
[100%] Built target stage3
Install the project...
-- Install configuration: "Debug"
```

**Comment by Vexu** [comment]
Fixed by #14471
